sections/footer.liquid

```diff
@@ -37,150 +37,163 @@
 {%- endstyle -%}
 
 <footer class="footer color-{{ section.settings.color_scheme }} gradient section-{{ section.id }}-padding">
+  {%-liquid 
+    assign no_social_icons = false
+    if settings.social_facebook_link == blank and settings.social_instagram_link == blank and settings.social_youtube_link == blank and settings.social_tiktok_link == blank and settings.social_twitter_link == blank and settings.social_pinterest_link == blank and settings.social_snapchat_link == blank and settings.social_tumblr_link == blank and settings.social_vimeo_link == blank
+      assign no_social_icons = true
+    endif
+
+    assign only_empty_brand = false
+    if section.blocks.size == 1 and section.blocks[0].type == 'brand_information' and no_social_icons and settings.brand_image == blank and settings.brand_headline == blank and settings.brand_description == blank and section.settings.newsletter_enable == false and section.settings.enable_follow_on_shop == false
```

Comment:
I decided to follow your suggestion although I'm not a fan of the almost-useless variable, but I agree that it's just too hard to know why we're checking for so many conditions.

Reply:
👍  I'm still trying to decide on the right balance of this myself. I feel ok about this one, mostly because if we needed to expand on this further approach to account for additional cases, I think there's a decent chance we'd end up needing these broken up more anyway to be reused in other similar conditionals. Hopefully with a more css-based solution we can avoid the issue he altogether though.